src/common/shared_config.h

```diff
@@ -7,32 +7,23 @@
 // bootloader version
 #define BOOTLOADER_VERSION_ADDRESS 0x0801FFFA
 
-// EEPROM firmware update flag
-#define FW_UPDATE_ENABLE  0xAA
-#define FW_UPDATE_DISABLE 0x00
-
-// pin PA13 state
-#define APPENDIX_FLAG_MASK 0x01
+typedef enum {
+    BT_APPENDIX_EXISTS = 0, /**< Appendix not broken, signature authentication needed */
+    BT_APPENDIX_BROKEN /**< Appendix broken, no signature authentication needed */
+} BT_APPENDIX_STATUS;
 
 #pragma pack(push)
 #pragma pack(1)
-typedef struct {
-    uint8_t fw_update_flag;
-    uint8_t model_specific_flags; // ~ "reserved1" originally
-    uint8_t reserved2;
-    uint8_t reserved3;
-    uint8_t reserved4;
-    uint8_t reserved5;
-    uint8_t reserved6;
-    uint8_t reserved7;
-    uint8_t reserved8;
-    uint8_t reserved9;
-    uint8_t reserved10;
-    uint8_t reserved11;
-    uint8_t reserved12;
-    uint8_t reserved13;
-    uint8_t reserved14;
-    uint8_t reserved15;
+
+typedef enum {
+    FW_UPDATE_ENABLE = 0xAA,
+    FW_UPDATE_DISABLE = 0xFF,
```

Comment:
Is 0xFF the default after clearing the flash memory area? We don't want the FW update disabled by default ... actually we are getting reports that fw update got disabled in FW 4.1. Can you explain it?

Reply:
The vales are taken from EEPROM in case of "update always" option, and in case of update on restart the values are from RAM, you meant stm flash area?